smarts/core/remote_agent_buffer.py

```diff
@@ -83,15 +89,17 @@ def destroy(self):
 
     def _spawn_local_zoo_worker(self, retries=3):
         local_port = find_free_port()
-        self._local_zoo_worker = Process(target=zoo_worker.listen, args=(local_port,))
+        self._local_zoo_worker = Process(
+            target=zoo_worker.listen, args=(local_port, self._authkey)
+        )
         self._local_zoo_worker.start()
 
         local_address = ("0.0.0.0", local_port)
 
         # Block until the local zoo server is accepting connections.
         for i in range(retries):
             try:
-                conn = Client(local_address, family="AF_INET")
+                conn = Client(local_address, family="AF_INET", authkey=self._authkey)
```

Comment:
The naming convention should not leak outside.

```suggestion
                conn = Client(local_address, family="AF_INET", authkey=self._auth_key)
```

Reply:
Changed internal `self.authkey` values to `self._auth_key`.